feat(perf): Update onPostBuild step to generate performance on-boarding docs

dashed opened the pull request:
This pull request extends the wizard onboarding documentation generation such that we generate Performance onboarding documentation content as sub-steps.

We aim to provide Performance onboarding content for JavaScript and React projects. They'll be shown in the product like this:

![Screen Shot 2022-05-13 at 3 10 10 PM](https://user-images.githubusercontent.com/139499/168374107-b8f6c675-0075-47a1-857c-8af4574d9574.png)

-----

Related to https://github.com/getsentry/sentry-docs/issues/3226

I intentionally did not provide any meta-documentation for how to write and contribute content to the Performance onboarding checklist at @imatwawana 's wishes; since we're only providing content for JavaScript and React platforms in the short-term to validate this feature.

AbhiPrasad commented on src/gatsby/onPostBuild.ts:
```diff
@@ -67,6 +67,37 @@ export default async ({ graphql }) => {
+      /^\/(?<platform>[^/]+)\/performance-onboarding\/(?<sub_platform>[^/]+)\/(?<step>[^/]+)\/$/
+    );
+    const { platform, sub_platform } = pathMatch.groups;
+    let step = pathMatch.groups.step;
+    step = String(step).replace(/\./g, "-");
```
```suggestion
    const step = String(pathMatch.groups.step).replace(/\./g, "-");
```

dashed commented:
@AbhiPrasad @lobsterkatie I've added you both as reviewers. Can either/both of you take a look at the Performance onboarding documentation content I've added in this PR, and see if I'm on the right track? 

These content will be displayed into the product in the form of a checklist; similar to how we serve wizard onboarding docs.

AbhiPrasad commented on src/gatsby/onPostBuild.ts:
```diff
@@ -67,6 +67,37 @@ export default async ({ graphql }) => {
+    const { platform, sub_platform } = pathMatch.groups;
+    let step = pathMatch.groups.step;
+    step = String(step).replace(/\./g, "-");
+    let sub = `performance-onboarding-${sub_platform}-${step}`;
+
+    if (platform === pathMatch.groups.sub_platform) {
+      sub = `performance-onboarding-${step}`;
+    }
```
```suggestion
    const sub = platform === sub_platform ? `performance-onboarding-${step}` : `performance-onboarding-${sub_platform}-${step}`;
```

AbhiPrasad commented on src/gatsby/onPostBuild.ts:
```diff
@@ -67,6 +67,37 @@ export default async ({ graphql }) => {
+  if (slug.includes("performance-onboarding")) {
+    const pathMatch = slug.match(
+      /^\/(?<platform>[^/]+)\/performance-onboarding\/(?<sub_platform>[^/]+)\/(?<step>[^/]+)\/$/
+    );
```
should we throw an error if we don't match here?

dashed replied:
@AbhiPrasad good catch. I've thrown an error here.

AbhiPrasad commented on src/wizard/javascript/performance-onboarding/javascript/3.verify.md:
````diff
@@ -0,0 +1,20 @@
+```javascript
+const transaction = Sentry.startTransaction({ name: "test-transaction" });
+const span = transaction.startChild({ op: "functionX" }); // This function returns a Span
+// functionCallX
````
```suggestion
// exampleFunctionCall();
```

AbhiPrasad commented on src/wizard/javascript/performance-onboarding/react/1.install.md:
````diff
@@ -0,0 +1,18 @@
+
+```bash
+# Using yarn
+yarn add @sentry/tracing
````
```suggestion
yarn add @sentry/react @sentry/tracing
```

AbhiPrasad commented on src/wizard/javascript/performance-onboarding/react/1.install.md:
```diff
@@ -0,0 +1,18 @@
+yarn add @sentry/tracing
+
+# Using npm
+npm install --save @sentry/tracing
```
```suggestion
npm install --save @sentry/react @sentry/tracing
```

AbhiPrasad reviewed (commented):
Can we add a note to `src/wizard` in a markdown file explaining how this all works?

imatwawana commented on src/wizard/javascript/performance-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,18 @@
+---
+name: JavaScript
+doc_link: https://docs.sentry.io/platforms/javascript/performance/
```
I'm not sure if this is correct. The enable tracing snippet on the performance page is slightly different. This identical snippet is found at: 
https://docs.sentry.io/platforms/javascript/#install

This applies to all of these files.

dashed replied:
@imatwawana I plan to use the `doc_link` attribute to place a link to the Performance setup pages on the Performance onboarding checklist.

Although the content is slightly different here, I wanted to provide the complete setup instructions without assuming the user has installed any dependencies beforehand.

imatwawana commented on src/wizard/javascript/performance-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,18 @@
+type: language
+---
+
+#### Install
```
Is there a specific reason we're using heading level 4 here? Is there a reason it isn't level 2? This applies to all of these files.

dashed replied:
@imatwawana This was intentional so that we have the "right" header sizing that fits well with the inserted checkboxes: 

<img width="450" alt="Screen Shot 2022-05-13 at 2 53 51 PM" src="https://user-images.githubusercontent.com/139499/168354231-94555cad-8f73-4322-af4e-926a98293381.png">

imatwawana commented on src/wizard/javascript/performance-onboarding/javascript/3.verify.md:
```diff
@@ -0,0 +1,20 @@
+
+#### Verify
+
+Verify that performance monitoring is working correctly by using our [automatic instrumentation](/javascript/performance/instrumentation/automatic-instrumentation/) by simply using your JavaScript application.
```
Removing repetition

```suggestion
Verify that performance monitoring is working correctly with our [automatic instrumentation](/javascript/performance/instrumentation/automatic-instrumentation/) by simply using your JavaScript application.
```

dashed replied:
Github isn't letting me apply this suggestion. So I added it manually.

imatwawana commented on src/wizard/javascript/performance-onboarding/react/3.verify.md:
```diff
@@ -0,0 +1,34 @@
+
+#### Verify
+
+Verify that performance monitoring is working correctly by using our [automatic instrumentation](/platforms/javascript/guides/react/performance/instrumentation/automatic-instrumentation/) by simply using your React application.
```
```suggestion
Verify that performance monitoring is working correctly with our [automatic instrumentation](/platforms/javascript/guides/react/performance/instrumentation/automatic-instrumentation/) by simply using your React application.
```

imatwawana commented on src/wizard/javascript/performance-onboarding/react/3.verify.md:
````diff
@@ -0,0 +1,34 @@
+transaction.finish(); // Finishing the transaction will send it to Sentry
+```
+
+In addition, `@sentry/react` exports a `withProfiler` higher order component that can be used to capture React related spans for specific React components:
````
```suggestion
In addition, `@sentry/react` exports a `withProfiler` higher order component that can be used to capture React-related spans for specific React components:
```

imatwawana reviewed (commented):
I've fixed some minor wording issues, but I've also left some questions.

dashed commented:
@imatwawana @AbhiPrasad Can either of you take a look at this pull request again? I'm fairly happy with the changes so far that I'd like to land this into `master` branch.

This change should have no impact to either the primary Sentry documentation website, or the Sentry product. 

The product feature that will consume the changes introduced in this pull request is being developed in https://github.com/getsentry/sentry/pull/33852. Thus, this pull request is a dependency of the developing Performance onboarding checklist feature.
